ugs-core/src/com/willwinder/universalgcodesender/TinyGController.java

```diff
@@ -254,14 +257,23 @@ private void sendInitCommands() {
         setStatusUpdateRate(getStatusUpdateRate());
     }
 
+    @Override
+    public void updateParserModalState(GcodeCommand command) {
+        // Prevent internal TinyG commands to update the parser modal state
+        if(!command.getCommandString().startsWith("{")) {
+            super.updateParserModalState(command);
+        }
+    }
+
     @Override
     public void performHomingCycle() throws Exception {
-        sendCommandImmediately(new GcodeCommand("G28.2 Z0 X0 Y0"));
+        sendCommandImmediately(new GcodeCommand("G28.2 X0 Y0 Z0"));
```

Comment:
As a data point, this won't make any functional difference.  The Z axis will be homed first anyway, as g2core regards that as being safer than homing it after the others.

Reply:
Sorry, this is from a bad merge. It's better having the Z probe first as it will reflect the correct order in g2core, will fix this soon!

Reply:
No worries at all. :smile: